py/frequenz/client/weather/_types.py

```diff
@@ -281,3 +281,85 @@ def to_ndarray_vlf(
             raise RuntimeError("Error processing forecast data") from e
 
         return array
+
+
+@dataclass(frozen=True)
+class HistoricalForecasts:
+    """Historical weather forecast data."""
+
+    _forecasts_pb: weather_pb2.GetHistoricalWeatherForecastResponse
+
+    @classmethod
+    def from_pb(
+        cls, forecasts: weather_pb2.GetHistoricalWeatherForecastResponse
+    ) -> HistoricalForecasts:
+        """Convert a protobuf Forecast message to Forecast object.
+
+        Args:
+            forecasts: protobuf message with historical forecast data.
+
+        Returns:
+            Forecast object corresponding to the protobuf message.
+        """
+        return cls(_forecasts_pb=forecasts)
+
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    def flatten(
+        self,
+    ) -> np.ndarray[
+        tuple[typing.Any, typing.Any, typing.Any, typing.Any, typing.Any, typing.Any],
+        np.dtype[np.float64],
+    ]:
+        """Flatten a Forecast object to a numpy array of tuples of data.
+
+        Returns:
+            Numpy array of tuples with the flattened forecast data.
+
+        Raises:
+            ValueError: If the forecasts data is missing or invalid.
+        """
+        # check for empty forecasts data
+        if not self._forecasts_pb.location_forecasts:
+            raise ValueError("Forecast data is missing or invalid.")
+
+        return flatten(list(self._forecasts_pb.location_forecasts))
```

Comment:
Because the function `flatten` takes as an argument list rather than repeated. I thought it is a better design decision that way as lists are simpler and more standard than repeated objects.